src/Mailchimp/Synchronisation/RequestBuilder.php

```diff
@@ -282,6 +289,24 @@ private function setZone(Zone $zone): void
         }
     }
 
+    public function setTeamCode(Adherent $adherent): self
+    {
+        if ($adherent->isParisResident()) {
+            $zones = $adherent->getZonesOfType(Zone::BOROUGH);
+        } else {
+            $zones = $adherent->getParentZonesOfType($adherent->isForeignResident() ? Zone::FOREIGN_DISTRICT : Zone::DEPARTMENT);
+        }
+
+        $count = \count($zones);
+        if ($count > 1) {
+            throw new \RuntimeException(\sprintf('Cannot find only one geo zone for Mailchimp for adherent with id "%s"', $adherent->getId()));
```

Comment:
this will crash the worker, maybe we can log the warning and set zone to null value?